### PR TITLE
refactor(input): replaces ring by border

### PR DESCRIPTION
**Pull request by soykje**
**TASK**: #1533 

### Description, Motivation and Context
As statued in main issue we want to replace borders that have been previously achieved through `box-shadow` (aka `ring-*`) by using good ol' `border`.

### Types of changes
- [x] 🧠 Refactor
- [x] 💄 Styles

### Screenshots - Animations
Capture d’écran du 2023-10-12 10-56-06
Capture d’écran du 2023-10-12 10-55-49

### CODE REVIEW

File: packages/components/input/src/Input.styles.ts [hunk 1]
```diff
@@ -26,15 +26,10 @@ export const inputStyles = cva(
...
-        neutral: [
-          'ring-outline',
-          'hover:ring-outline-high',
-          'focus:ring-outline-high',
-          'disabled:ring-outline',
-        ],
-        success: ['ring-success', 'ring-success'],
-        alert: ['ring-alert', 'ring-alert'],
-        error: ['ring-error', 'ring-error'],
+        neutral: ['border-outline', 'hover:border-outline-high', 'disabled:border-outline'],
```
Comment (acd02) [inline packages/components/input/src/Input.styles.ts:29]: Is it normal that only the neutral intent has its own **disabled** styles?

**Comment by soykje** [comment]
Not sure if this is what we want...
image

I'll look for updates from design + lead dev :+1:

**Comment by soykje** [comment]
After discussions with both @JeremiasUX and @andresz1 we agreed that validation states (error, succes, alert) shouldn't occur at the same time as `disabled` state. It is all about user feedback, which shouldn't happen on disabled input... So small style reinforcements have been done "just in case" :+1: But no addition to stories about this very edge case, that could only happen on bad implementation.

File: packages/components/input/src/Input.styles.ts (hunk 1 above)
Comment (soykje) [reply packages/components/input/src/Input.styles.ts]: See https://github.com/adevinta/spark/pull/1560#issuecomment-1757364903

**Comment by soykje** [comment]
After last discussions, the focus styles for this component will only be black ones :+1:

**Review by acd02 — APPROVED** [review]

**Review by andresin87 — APPROVED** [review]

**Review by turolopezsanabria — APPROVED** [review]
👍🏼 Design check ok, thnx!

**Pull request merged**